haystack/components/audio/whisper_remote.py

```diff
@@ -62,33 +58,20 @@ def __init__(
             temperature until certain thresholds are hit.
         """
 
-        # if the user does not provide the API key, check if it is set in the module client
-        api_key = api_key or openai.api_key
-        if api_key is None:
-            try:
-                api_key = os.environ["OPENAI_API_KEY"]
-            except KeyError as e:
-                raise ValueError(
-                    "RemoteWhisperTranscriber expects an OpenAI API key. "
-                    "Set the OPENAI_API_KEY environment variable (recommended) or pass it explicitly."
-                ) from e
-        openai.api_key = api_key
-
         self.organization = organization
         self.model_name = model_name
         self.api_base_url = api_base_url
 
         # Only response_format = "json" is supported
         whisper_params = kwargs
-        if whisper_params.get("response_format") != "json":
+        response_format = whisper_params.get("response_format", "json")
+        if response_format != "json":
             logger.warning(
                 "RemoteWhisperTranscriber only supports 'response_format: json'. This parameter will be overwritten."
             )
         whisper_params["response_format"] = "json"
         self.whisper_params = whisper_params
-
-        if organization is not None:
-            openai.organization = organization
+        self.client = OpenAI(api_key=api_key, organization=organization, base_url=api_base_url)
```

Comment:
I made sure we instantiated all OpenAI instances with these three parameters. Perhaps soon we'll have other providers of whisper (other base_url). I found organization important for production (identify different subgroups and how much they spend e.g.)